skopt/utils.py

```diff
@@ -292,3 +295,50 @@ def cook_estimator(base_estimator, space=None, **kwargs):
 
     base_estimator.set_params(**kwargs)
     return base_estimator
+
+def transform_dims(dimensions, estimator="GP"):
+    """
+    Transforms space for Gaussian processes.
+
+    Parameters
+    ----------
+    * `dimensions` [list, shape=(n_dims,)]:
+        List of search space dimensions.
+        Each search dimension can be defined either as
+
+        - a `(upper_bound, lower_bound)` tuple (for `Real` or `Integer`
+          dimensions),
+        - a `(upper_bound, lower_bound, "prior")` tuple (for `Real`
+          dimensions),
+        - as a list of categories (for `Categorical` dimensions), or
+        - an instance of a `Dimension` object (`Real`, `Integer` or
+          `Categorical`).
+
+         NOTE: The upper and lower bounds are inclusive for `Integer`
+         dimensions.
+    * `estimator` [str, ]:
+        optimization estimator to be used.
+        each estimator may use different transformations for dimensions.
+    """
+    if estimator == "GP":
```

Comment:
The alternative might be that we assume the user knows what they are doing when they call this and we just always noramlise the dimensions according to the below recipe. Leaving it to the user to decide if they want to call this or not. Think this is my favourite option at the moment.